r-package/grf/R/tuning.R

```diff
@@ -0,0 +1,174 @@
+#' Tune a forests
+#'
+#' Finds the optimal parameters to be used in training a regression or causal forest.
+#'
+#' @param data The data arguments (output from create_data_matrices) for the forest.
+#' @param args The remaining call arguments for the forest.
+#' @param tune.parameters The vector of parameter names to tune.
+#' @param forest.type forest.type The type of forest to tune.
+#' @param num.fit.trees The number of trees in each 'mini forest' used to fit the tuning model.
+#' @param num.fit.reps The number of forests used to fit the tuning model.
+#' @param num.optimize.reps The number of random parameter values considered when using the model
+#'  to select the optimal parameters.
+#'
+#' @return tuning output
+#' @importFrom stats sd runif
+#' @importFrom utils capture.output
+tune_forest <- function(X,
+                        data,
+                        args,
+                        tune.parameters,
+                        forest.type = c("regression_forest", "causal_forest"),
+                        num.fit.trees = 10,
+                        num.fit.reps = 100,
+                        num.optimize.reps = 1000) {
+  forest.type <- match.arg(forest.type)
+
+  if (forest.type == "regression_forest") {
+    fit <- regression_train
+    predict_oob <- regression_predict_oob
+    predict.data.args <- c("train.matrix", "sparse.train.matrix", "outcome.index")
+    defaults <- formals(regression_forest)
+  } else if (forest.type == "causal_forest") {
+    fit <- causal_train
+    predict_oob <- causal_predict_oob
+    predict.data.args <- c("train.matrix", "sparse.train.matrix", "outcome.index", "treatment.index")
+    defaults <- formals(causal_forest)
+  }
```

Comment:
What happens here if I forget to input a `forest.type`?

Reply:
```
> match.arg(c("regression_forest", "causal_forest"), c("regression_forest", "causal_forest"))
[1] "regression_forest"
```